File: python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  silnia.py
# Oblicznie potęgi liczby naturalnej

def silnia_it(n):
    """
    n! = 1 * 2 * ... * n
    """
    wynik = 1
    for i in range(1, n+1):
        wynik *= i
    return wynik

def main(args):
    n = 3
    print("Silnia liczby {} wynosi {}.".
           format(n, silnia_it(n)))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>

using namespace std;

int silnia_it(int a){
    int wynik = 1;
    for(int i=1; i<a+1; i++) wynik *= i;
    return wynik;
}

//5!
// silnia_re(5)
// silnia_re(4) * 5
// silnia_re(3) * 4
// silnia_re(2) * 3
// silnia_re(1) * 2
// silnia_re(0) * 1
// 1
// 1 * 1
// 1 * 2
// 2 * 3
// 6 * 4
// 24 * 5
// 120

int silnia_re(int a){
    if(a==1) return 1;
    return a*silnia_re(a-1);
}

int main(int argc, char **argv){
    int a;
    cout << "Podaj podstawę: ";
    cin >> a;
	cout << "Silnia liczby " << a << " wynosi " << silnia_it(a) << endl;
    cout << "Silnia liczby " << a << " wynosi " << silnia_re(a) << endl;
	return 0;
}

